src/dmd/parse.d

```diff
@@ -247,9 +248,23 @@ final class Parser(AST) : Lexer
             /* Create a pseudo-filename for the mixin string, as it may not even exist
              * in the source file.
              */
-            char* filename = cast(char*)mem.xmalloc(strlen(loc.filename) + 7 + (loc.linnum).sizeof * 3 + 1);
-            sprintf(filename, "%s-mixin-%d", loc.filename, cast(int)loc.linnum);
-            scanloc.filename = filename;
+            auto len = strlen(loc.filename) + 7 + (loc.linnum).sizeof * 3 + 1;
+            char[] filename = cast(char[])mem.xmalloc(len)[0 .. len];
+            sprintf(filename.ptr, "%s-mixin-%d", loc.filename, cast(int)loc.linnum);
+
+            // If we are debugging mixins, write `input` out to a file so that the debugger has some
+            // source to look at.
+
+            if (global.params.debugMixins)
+            {
+                auto tmp = FileName.combine(global.params.objdir,filename.ptr);
+                filename = cast(char[])tmp[0 .. strlen(tmp)];
+                auto f = File.create(filename.ptr);
```

Comment:
Use `new File(filename.ptr)`. `create` is a C++ wrapper.

Reply:
Actually doesn't need to even be `new`'d